pkg/kubelet/nodestatus/setters.go

```diff
@@ -481,20 +483,49 @@ func GoRuntime() Setter {
 }
 
 // RuntimeClasses returns a Setter that sets RuntimeClasses on the node.
-func RuntimeClasses(fn func() []kubecontainer.RuntimeHandler) Setter {
+func RuntimeClasses(
+	rcManager *runtimeclass.Manager,
+	handlersFunc func() []kubecontainer.RuntimeHandler,
+) Setter {
 	return func(ctx context.Context, node *v1.Node) error {
-		if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
+		if rcManager == nil || !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
 			return nil
 		}
-		handlers := fn()
-		node.Status.RuntimeClasses = make([]v1.NodeRuntimeClass, len(handlers))
-		for i, h := range handlers {
-			node.Status.RuntimeClasses[i] = v1.NodeRuntimeClass{
-				Name: h.Name,
+		handlers := handlersFunc()
+		handlersMap := make(map[string]kubecontainer.RuntimeHandler, len(handlers))
+		for _, h := range handlers {
+			handlersMap[h.Name] = h
+		}
+
+		classes, err := rcManager.Lister().List(labels.Everything())
+		if err != nil {
+			return fmt.Errorf("failed to get the list of the runtime classes: %w", err)
+		}
+		node.Status.RuntimeClasses = []v1.NodeRuntimeClass{}
+		handler, ok := handlersMap[""]
+		if !ok {
+			return fmt.Errorf("failed to look up the default runtime handler")
+		}
+		node.Status.RuntimeClasses = append(node.Status.RuntimeClasses, v1.NodeRuntimeClass{
+			Name: "", // for the default runtime class
+			Features: &v1.NodeRuntimeClassFeatures{
+				RecursiveReadOnlyMounts: &handler.SupportsRecursiveReadOnlyMounts,
+			},
+		})
+
+		for _, class := range classes {
+			handler, ok = handlersMap[class.Handler]
+			if !ok {
+				klog.ErrorS(nil, "Failed to look up the runtime handler", "runtimeClassName", class.Name,
```

Comment:
This is not necessarily an error. A use case is that there is a dedicated node pool for running sandboxed containers, and a certain RuntimeClass/handler is only available in that node pool. In fact, this is a supported use case through the Scheduling mix-in on the RuntimeClass: https://github.com/kubernetes/api/blob/96558b97565e2a9bed0b5cb84db3b09b71d20172/node/v1/types.go#L65

Reply:
The code is removed